pkg/controller/apmserver/apmserver_controller.go

```diff
@@ -235,7 +235,18 @@ func (r *ReconcileApmServer) doReconcile(ctx context.Context, request reconcile.
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	results := apmcerts.Reconcile(ctx, r, as, []corev1.Service{*svc}, r.CACertRotation, r.CertRotation)
+	_, results := certificates.ReconcileCAAndHTTPCerts(
```

Comment:
Previously this was a noop because `apmcerts.Reconcile` returned immediately if the self-signed certificate was disabled. Can't we just move that check up here and avoid calling `ReconcileCAAndHTTPCerts` if TLS is disabled?

Reply:
we can do that, but existing secrets are not removed, which is a bug IMO.
I think it's still worth exploring though 👍

Reply:
Added an option to garbage collect secrets if TLS is disabled. Used by the Kibana, APMServer and Enterprise Search controllers.